tests/System.IO.Pipelines.Tests/UnownedBufferReaderFacts.cs

```diff
@@ -333,9 +326,21 @@ public class UnownedBufferReaderFacts
                 Assert.Equal("Hello ", Encoding.UTF8.GetString(preserved.Buffer.ToArray()));
             }
 
+            EnsureSpanDisposed(preserved.Buffer.First);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
```

Comment:
I still don't understand why it is necessary. 
If the method call can get optimized out, won't this get optimized too:
https://github.com/dotnet/corefxlab/pull/1432/files#diff-920b95938a7bd6c56992737a476f216eL326
```C#
// Make sure we can acccess the span
 var span = buffer.First.Span;
```